test/edu/ucsb/nceas/metacat/restservice/v2/MNResourceHandlerTest.java

```diff
@@ -0,0 +1,382 @@
+package edu.ucsb.nceas.metacat.restservice.v2;
+
+import org.apache.wicket.protocol.http.mock.MockHttpServletRequest;
+import org.apache.wicket.protocol.http.mock.MockHttpServletResponse;
+import org.apache.wicket.protocol.http.mock.MockHttpSession;
+import org.apache.wicket.protocol.http.mock.MockServletContext;
+import org.dataone.service.types.v1.Identifier;
+import org.dataone.service.types.v1.Session;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.List;
+import java.util.Vector;
+
+import edu.ucsb.nceas.LeanTestUtils;
+import edu.ucsb.nceas.metacat.dataone.MNodeService;
+
+
+
+/**
+ * A test class of the MNResourceHandler class
+ * @author tao
+ *
+ */
+
+public class MNResourceHandlerTest {
+
+    /**HTTP Verb GET*/
+    protected static final byte GET = 1;
+    /**HTTP Verb POST*/
+    protected static final byte POST = 2;
+    /**HTTP Verb PUT*/
+    protected static final byte PUT = 3;
+    /**HTTP Verb DELETE*/
+    protected static final byte DELETE = 4;
+    /**HTTP Verb HEAD*/
+    protected static final byte HEAD = 5;
+
+    private MockHttpServletRequest request = null;
+    private MockServletContext context = null;
+    private MNResourceHandler resourceHandler = null; 
+
+    private static final String PATH = "/";
+    private static final String ENCODED_PID =
+         "http%3A%2F%2Fdx.doi.org%2F10.5061%2Fdryad.12%3Fver%3D2017-08-29T11%3A52%3A08.075-05%3A00";
+    private static final String DECODED_PID =
+                             "http://dx.doi.org/10.5061/dryad.12?ver=2017-08-29T11:52:08.075-05:00";
+    private static final String URN_PID = "urn:uuid:de8528af-3636-44e7-8db5-ce5c6ac95770";
+    private static final String DOI1 = "doi:10.5072/FK2FR01T7X1";
+    private static final String DOI2 = "doi:10.5072/FK2T155C0Q3";
+    private static final String EML2_NAMESPACE = "eml://ecoinformatics.org/eml-2.0.0";
+    private static final String EML201_NAMESPACE = "eml://ecoinformatics.org/eml-2.0.1";
+
+    /**
+     * Setup
+     */
+    @Before
+    public void setUp() throws Exception {
+        LeanTestUtils.initializePropertyService(LeanTestUtils.PropertiesMode.UNIT_TEST);
+        context = new MockServletContext(null, PATH);
+        //@TODO The filter seems not working
```

Comment:
unresolved TODO?

Reply:
I haven't figured it out. So far the default filter works for our testing. I can image it will have some issues for the sql query test. So I defer it. Do you have any idea?

Reply:
I would suggest making that more clear. For example, if it doesn't work, can you just delete the `context.addFilter("d1Filter"...` line for now? 

And maybe update this comment to clarify what you said above? e.g. instead of a TODO, maybe it could be: 

```java
// The default filter works for our testing. 
// Note, however, that the following filter does NOT seem to work:
//  context.addFilter("d1Filter", "edu.ucsb.nceas.metacat.restservice.D1URLFilter");
// This is OK for now, but may become a problem for the sql query test" 
```